nominal/core/dataset.py

```diff
@@ -185,6 +188,40 @@ def get_channels(
             else:
                 next_page_token = response.next_page_token
 
+    def to_pandas(self, channel_exact_match: Sequence[str] = (), channel_fuzzy_search_text: str = "") -> pd.DataFrame:
+        """Download a dataset to a pandas dataframe, optionally filtering for only specific channels of the dataset.
+        Args:
+            channel_exact_match: Filter the returned channels to those whose names match all provided strings (case insensitive).
+                For example, a channel named 'engine_turbine_rpm' would match against ['engine', 'turbine', 'rpm'],
+                whereas a channel named 'engine_turbine_flowrate' would not!
+            channel_fuzzy_search_text: Filters the returned channels to those whose names fuzzily match the provided string.
+        Returns:
+            A pandas dataframe whose index is the timestamp of the data, and column names match those of the selected channels.
+
+        Example:
+        ```
+        import nominal as nm
+
+        rid = "..." # Taken from the UI or via the SDK
+        dataset = nm.get_dataset(rid)
+        s = dataset.to_pandas()
+        print("index:", s.index, "index mean:", s.index.mean())
+        ```
+        """
+        # We concurrently download all channels for faster performance
```

Comment:
The code path to download a single channel is actually parameterized for a list of channels as well. The backend stitches it all together into a single csv, so we don't actually need to invoke this in parallel per channel and concat them client side

Reply:
but I think performance wise, we'd rather download n channels at the same time than ask the backend for a giant file, because we can get each channel concurrently since the network request is non-blocking.

Reply:
The backend is already creating the CSV for each channel as well, so I don't think it's likely it will be faster client-side. we can perf test tho?

Reply:
nice, thanks for that info. You are definitely right here, we don't need to test. The only thing to note here is that we are losing the "time to first response" UX since we have to wait for the entire dataset to download, but we can handle that separately.

updated to leverage that logic